### PR TITLE
Support up-to-date `botocore`

### PR DESCRIPTION
**Pull request by jakob-keller**
### Description of Change
This PR intends to improve general compatibility of `aiobotocore` within the Python ecosystem by relaxing the dependency specification of `botocore`, as well as `boto3` and `awscli`.

### Assumptions
[Upstream diff](https://github.com/boto/botocore/compare/1.31.64..1.31.70) does not contain any changes that require adjustments to the `aiobotocore` codebase. This allows for the proposed widening of the `botocore` dependency specification.

### Checklist for All Submissions
* [x] I have added change info to [CHANGES.rst](https://github.com/aio-libs/aiobotocore/blob/master/CHANGES.rst)

### Checklist when updating botocore and/or aiohttp versions

* [x] I have read and followed [CONTRIBUTING.rst](https://github.com/aio-libs/aiobotocore/blob/master/CONTRIBUTING.rst#how-to-upgrade-botocore)
* [x] I have updated test_patches.py where/if appropriate (also check if no changes necessary)
* [x] I have ensured that the awscli/boto3 versions match the updated botocore version

### CODE REVIEW

**Comment by jakob-keller** [comment]
I intend to monitor `botocore` releases and regularly push updates until it is worth merging this PR.

File: CHANGES.rst [hunk 1]
```diff
@@ -1,9 +1,10 @@
 Changes
 -------
 
-2.7.1 (2023-10-19)
+2.7.1 (2023-10-21)
```
[the remark this answers is not in the document]
Comment (thehesiod) [reply CHANGES.rst:4]: actually I'll do this in master

**Comment by thehesiod** [comment]
ok let me know when you want to merge

**Comment by thehesiod** [comment]
want to convert to draft meanwhile?

**Comment by jakob-keller** [comment]
I suggest we merge now and extend support to include `botocore==1.31.70`.

`botocore==1.31.71` might be a little bit more involved due to https://github.com/boto/botocore/pull/3037.

**Pull request merged**